### PR TITLE
Clean up unwanted trailing spaces

### PR DESCRIPTION
**Pull request by dragonmaus**
Also make EditorConfig remove them by default outside of explicit exemptions.

I did not touch:
- lines in data and code files that contain only whitespace, as these are a deliberate stylistic choice by the devs; and
- lines in markdown files that have semantically significant trailing double-spaces.

(This is a one-off sort of commit that doesn't fit any of the template's categories, so please pardon my not using it.)

- [x] This PR does not modify, remove, or add any copyright attributions or claims.
- [x] This PR does not affect in any way the compilation or behaviour of the game.

### CODE REVIEW

**Comment by Zitchas** [comment]
Quite the clean up! I haven't looked through it all, but the parts I did look through look good.

**Comment by dragonmaus** [comment]
> 
> 
> Quite the clean up! I haven't looked through it all, but the parts I did look through look good.

To be honest, this is just the lowest-possible hanging fruit. A proper cleanup would be a lot more invasive. 😁

Though to be fair, the ES codebase seems fairly well maintained, and I haven't noticed anything really egregious yet.

File: XCode/Images.xcassets/AppIcon.appiconset/Contents.json [hunk 1]
```diff
@@ -55,4 +55,4 @@
     "version" : 1,
     "author" : "xcode"
   }
-}
\ No newline at end of file
+}
```
Comment (Amazinite) [inline XCode/Images.xcassets/AppIcon.appiconset/Contents.json]: Don't eat the newline at the end of files. (Luckily it looks like this is the only one.)

Comment (dragonmaus) [reply XCode/Images.xcassets/AppIcon.appiconset/Contents.json]: That's actually adding a newline. Unless I'm misinterpreting what is wanted here?

**Review by tehhowch — COMMENTED** [review]
One change

**Pull request merged**